packages/admin-app/src/pages/LoginPage.tsx

```diff
@@ -1,16 +1,117 @@
-import { Box } from '@material-ui/core';
-import React from 'react';
+import {
+    Body_login_for_access_token_api_v1_auth_login_post,
+    AuthClient,
+    UrlGetter,
+    ErrorMessage,
+} from '@electionguard/api-client';
+import { Button, Container, InputAdornment, makeStyles, TextField } from '@material-ui/core';
+import { AccountCircle, Lock } from '@material-ui/icons';
+import React, { useState } from 'react';
 
-export const LoginPage: React.FC = () => (
-    <Box
-        height="100%"
-        display="flex"
-        flexDirection="column"
-        justifyContent="center"
-        alignItems="center"
-    >
-        Login
-    </Box>
-);
+const useStyles = makeStyles((theme) => ({
+    root: {
+        height: '100%',
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center',
+    },
+    errorMessage: {
+        color: 'red',
+        marginBottom: theme.spacing(3),
+    },
+    text: {
+        marginBottom: theme.spacing(3),
+        width: '100%',
+    },
+    submit: {
+        marginTop: theme.spacing(3),
+        width: '100%',
+    },
+}));
+
+export interface LoginPageProps {
+    setToken: (token: string) => void;
+}
+
+export const LoginPage: React.FC<LoginPageProps> = ({ setToken }) => {
+    const classes = useStyles();
+    const [username, setUserName] = useState('');
+    const [password, setPassword] = useState('');
+    const [result, setResult] = useState<string>();
+
+    const handleSubmit: React.FormEventHandler<HTMLFormElement> = async (e) => {
+        e.preventDefault();
+        const url = UrlGetter.GetUrl();
+        const authClient = new AuthClient(url);
+        const loginParams = new Body_login_for_access_token_api_v1_auth_login_post();
```

Comment:
Ugly generated name, I'll see if I can't clean this up to avoid the need for this

Reply:
I was unable to easily fix it by playing with the swagger file.  If I have nswag generate DTO types as interfaces rather than classes, then this code changes from `new Body_login_for_access_token_api_v1_auth_login_post()` with a lot of setters to just:

```
        const loginParams = {
            username,
            password,
            grant_type: 'password',
            scope: 'admin',
            client_id: 'electionguard-default-client-id',
            client_secret: 'electionguard-default-client-secret',
        } as Body_login_for_access_token_api_v1_auth_login_post;
```

Which I personally find more attractive.  Anyone have any preferences on classes vs interfaces for DTO's?